StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx

```diff
@@ -0,0 +1,598 @@
+			continue;
+		}
+
+		// Define apv Id to form full channel id later
+		int apvElecId = (rdo - 1) * kFstNumArmsPerRdo * kFstNumApvsPerArm * kFstNumApvChannels + arm * kFstNumApvsPerArm * kFstNumApvChannels + apv * kFstNumApvChannels;
+
+		// Loop over the data in this APV to get raw hit info. (channel, timebin, adc)
+		for (StRtsTable::iterator it = rts_tbl->begin(); it != rts_tbl->end(); it++) {
+			// channel info.
+			fgt_adc_t *f = (fgt_adc_t *)*it;
+			Int_t channel   = f->ch;  //channel index  0, 1, ..., 127
+			Int_t adc       = f->adc; //adc
+			Short_t timebin = f->tb;  //time bin
+			LOG_DEBUG << "channel: " << channel << "   adc: " << adc << "  time bin: " << timebin << endm;
+
+			flag = 0;
+
+			if ((dataFlag == mADCdata) && (adc < 0 || adc >= kFstMaxAdc))  flag = 1;
+			if (channel < 0 || channel >= kFstNumApvChannels)              flag = 1;
+			if (timebin < 0 || timebin >= ntimebin)                        flag = 1;
+
+			if (flag == 1) {
+				LOG_INFO << "Corrupt data channel: " << channel << " tbin: " << timebin << " adc: " << adc << endm;
+				continue;
+			}
+
+			signalUnCorrected[channel][timebin] = adc;
+			if(adc>0) nRawAdcFromData++;
+
+			if ( !mIsCaliMode )        {
+				Int_t elecId = apvElecId + channel;
+
+				if (elecId < 0 || elecId >= kFstNumElecIds) {
+					LOG_INFO << "Wrong elecId: " << elecId  << endm;
+					continue;
+				}
+
+				// This is where we get the simulated hits from the simu container if it is available
+				// and merge with real data ADC values
+				if (mFstCollectionSimuPtr)
+				{
+					Int_t geoId = mMappingVec[elecId];
+					Int_t wedge = 1 + geoId / (kFstNumInnerSensorsPerWedge * kFstNumStripsPerInnerSensor + kFstNumOuterSensorsPerWedge * kFstNumStripsPerOuterSensor);
+
+					StFstRawHitCollection *rawHitCollectionSimuPtr = mFstCollectionSimuPtr->getRawHitCollection(wedge-1);
+					if(rawHitCollectionSimuPtr)
+					{
+						StFstRawHit * rawHitSimu = rawHitCollectionSimuPtr->getRawHit(elecId);
+						idTruth[channel] = 0;
+						if(rawHitSimu->getCharge(timebin) > 0) {  // a valid MC hit
+							signalUnCorrected[channel][timebin] += rawHitSimu->getCharge(timebin);
+							idTruth[channel] = rawHitSimu->getIdTruth();
+						}
+					}
+				}
+
+				if ( dataFlag == mADCdata ) { // non-ZS data
+					signalCorrected[channel][timebin] = signalUnCorrected[channel][timebin] - mPedVec[elecId][timebin];
+
+					// exclude signal-related channels for common mode noise calculation
+					if ( (signalCorrected[channel][timebin] > (-mCmnCut)*mTotRmsVec[elecId][timebin]) &&
+							(signalCorrected[channel][timebin] <   mCmnCut *mTotRmsVec[elecId][timebin]) )
+					{
+						Int_t geoId = mMappingVec[elecId];
+						int rstrip = (geoId % (kFstNumInnerSensorsPerWedge * kFstNumStripsPerInnerSensor + kFstNumOuterSensorsPerWedge * kFstNumStripsPerOuterSensor))/kFstNumPhiSegPerWedge;
+						for (int rindex = 0; rindex < kFstNumRStripsPerWedge; rindex++){
+							if(rindex == rstrip){
```

Comment:
You already know the value of rstrip, why do you have to find it in a loop?

Reply:
fixed by using rstrip directly and removed the loop.